tests/psa-client-server/psasim/src/psa_sim_client_config_adjust.h

```diff
@@ -0,0 +1,14 @@
+/**
+ * \file psa_sim_client_config_adjust.h
+ *
+ * \brief User config file for client-side code to be run under PSA simulator
+ */
+
+/*
+ *  Copyright The Mbed TLS Contributors
+ *  SPDX-License-Identifier: Apache-2.0 OR GPL-2.0-or-later
+ */
+
+#define MBEDTLS_PSA_CRYPTO_CLIENT
+#undef MBEDTLS_PSA_CRYPTO_C
+#undef MBEDTLS_PSA_CRYPTO_STORAGE_C
```

Comment:
Well, `all.sh` is definitely not doing it when building the client pieces with this `Makefile` - I had an assert on the size of the operation_t (that it was the size of a handle), and it fired without this, and was quiet with this.

It may well be possible to do this in `all.sh`, but I wanted to get us up and running again!

We don't have to merge this PR; bits of it can go into others

Reply:
Ok, let me pick only serializing/deserializing commits from this PR then into mine and see if that fix the issues with tests.

Reply:
I've reverted the commit that made this change, and added the commit from you @valeriosetti that does this in `all.sh` - thanks